sdk/core/core-rest-pipeline/src/policies/throttlingRetryPolicy.ts

```diff
@@ -22,21 +27,23 @@ export function throttlingRetryPolicy(): PipelinePolicy {
   return {
     name: throttlingRetryPolicyName,
     async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
-      const response = await next(request);
-      if (response.status !== 429) {
-        return response;
-      }
+      let response = await next(request);
 
-      const retryAfterHeader = response.headers.get("Retry-After");
-
-      if (retryAfterHeader) {
+      for (let count = 0; count <= DEFAULT_CLIENT_MAX_RETRY_COUNT; count++) {
+        if (response.status !== 429 && response.status !== 503) {
+          return response;
+        }
+        const retryAfterHeader = response.headers.get("Retry-After");
+        if (!retryAfterHeader) {
+          break;
```

Comment:
why have `break` here and below, but the direct return above? Breaking out of the for loop simply returns anyway, so we might as well have returns in all three places